providers/default.rb

```diff
@@ -0,0 +1,80 @@
+#
+# Cookbook Name:: nodejs-webapp
+# Provider:: default
+#
+# Copyright 2015, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+action :install do
+  # Use the git recipe to install git
+  run_context.include_recipe 'git'
+  run_context.include_recipe 'build-essential'
+  run_context.include_recipe 'pm2'
+
+  magic_shell_environment 'PATH' do
+    value '/usr/local/bin:$PATH'
+  end
+
+  # If the new_resource.path is nil set the install path to
+  # `/opt/name_attribute`
+  path = new_resource.path || "/opt/#{new_resource.name}"
+
+  if new_resource.create_user
+    group new_resource.group do
+      action :create
+    end
+    user new_resource.user do
+      action :create
+      gid new_resource.group
+    end
+  end
+
+  directory path do
```

Comment:
They could still write their bad files to somewhere else on disk. It's not a bad thought, but you need to take it a few steps further to really make it work.